_docs/advanced-configurations/programming-escs-with-passthrough.md

```diff
@@ -0,0 +1,25 @@
+---
+title: "Programming ESCs with Passthrough"
+excerpt: "How to use BLHeliConfigurator or BLHeliSuite with dRonin"
+---
+
+{% include callout type="danger" title="Remove props before ESC programming!" text="ESC programming generally requires battery power to be provided to the ESCs, and for unusual signals to be sent down the control line to the ESC.  Motors can accidentally beign spinning.  Ensure that **ALL PROPS** are removed from the vehicle prior to ESC programming and that wires and fingertips will not become entangled in motors if they begin to spin." %}
+
+dRonin now supports programming ESCs using 4WIF-style passthrough.  The "MSP" module emulates the MultiWii/CleanFlight/BetaFlight configuration interface.  BLHeli, BLHeli_S , BLHeli32, and SimonK ESCs can be reprogrammed.
+
+First, it's important to configure the USB virtual com port to speak the MSP protocol.  On the hardware configuration page, select that port as "MSP" and turn off any other MSP ports used for on-screen displays.  (This configuration can be reversed after ESC programming to save resources and re-enable character OSD support).  Restart the flight controller.
```

Comment:
"turn off any other MSP ports used for on-screen displays" tells me not to turn off MSP ports used for things that aren't OSDs too (otherwise why would you be specific?).

```First, it's important to configure the USB virtual com port to speak the MSP protocol.  On the hardware configuration page, select that port as "MSP" and turn off any other MSP ports.  (This configuration can be reversed after ESC programming to save resources and re-enable OSDs etc.).```